### ISSUE TITLE
Grouping by column name

### ISSUE THREAD

**Issue by nbrown02** [MicrosoftDocs/vsts-docs#6928]
Is it possible to group by column name? We have the 'committed' state mapped to multiple columns (In Dev / Ready for Test / In Test) so this breakdown would be useful...

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8719e400-4445-59a4-c68e-628243710c36
* Version Independent ID: d6a76999-81c0-33cb-47a5-9ca4d8be80c3
* Content: [Feature Progress rollup sample Power BI report - Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/report/powerbi/sample-boards-featureprogress?view=azure-devops&tabs=powerbi)
* Content Source: [docs/report/powerbi/sample-boards-featureprogress.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/report/powerbi/sample-boards-featureprogress.md)
* Product: **devops**
* Technology: **devops-analytics**
* GitHub Login: @KathrynEE
* Microsoft Alias: **kaelli**

**Comment by msebolt** [comment]
@nbrown02 Thank you for the question.

Please refer to Power BI documentation for information on how to customize the reports, for instance, the [Group rows](https://docs.microsoft.com/power-bi/desktop-common-query-tasks#group-rows) topic seems similar to what you are asking.

**Comment by nbrown02** [comment]
Thanks - I guess let me rephrase my question :)

Is the ColumnName field within the WorkItems table/. If not, is it possible to recreate this chart using an alternative query so it uses ColumnName rather than State?

**Comment by msebolt** [comment]
Thank you for the update.

In this case you'll need to use an additional PowerBI query to wrap the WorkItems table query and access the column name that way, perhaps the [ColumnNames](https://docs.microsoft.com/powerquery-m/table-columnnames) property? 

To see work item fields available, please refer to the **Azure DevOps** [data model](https://docs.microsoft.com/azure/devops/report/extend-analytics/data-model-analytics-service).

**Comment by nbrown02** [comment]
Ok - sadly doesn't look like I'm getting much of a clear answer :(

The above shows how to group child items by StateCategory, all I'm really try to find out is if/how this could be done to group child items by Board Column...

**Comment by msebolt** [comment]
Apologies for the delay and not fully understanding the problem.

Typically, grouping by Board Column would be the same as grouping by State, and can be accomplished with default PowerBI sorting along with an OData query.

For example, I sort by State (BoardColumn) in the following table:

![Capture](https://user-images.githubusercontent.com/6675884/75727819-c9e26d80-5c9a-11ea-96aa-febdf20f6221.PNG)

...using OData query:

```
let
   Source = OData.Feed ("http://localhost/<collection>/<project>/_odata/v1.0/WorkItems?"
        ,null, [Implementation="2.0",OmitValues = ODataOmitValues.Nulls,ODataVersion = 4]) 
in
    Source
```

**Comment by nbrown02** [comment]
No worries - Yup I have that (using State) already with a number of teams - but its not possible using BoardColumn I don’t think? (Hoping you prove me wrong)

**Comment by msebolt** [comment]
Thank you for your understanding, I guess I was assuming by "BoardColumn" you were referring to the **State** column or any of these other columns (see below),

![75727819-c9e26d80-5c9a-11ea-96aa-febdf20f6221](https://user-images.githubusercontent.com/6675884/77461879-7c03e580-6dc0-11ea-843c-c581283cbfb4.png)

If not, perhaps [contacting the product team](https://developercommunity.visualstudio.com/spaces/21/index.html) would be the best option at this point.  Our apologies for not offering a better solution.

Closing the issue now.
